selectrum.el

```diff
@@ -825,39 +860,71 @@ currently displayed candidates."
             (window-resize
              win (- dheight wheight) nil nil 'pixelwise)))))))
 
-(defun selectrum--first-lines (candidates)
+(defun selectrum--ensure-single-lines (candidates)
   "Return list of single line CANDIDATES.
-Multiline canidates are merged into a single line."
-  (let ((onelines ()))
-    (dolist (cand candidates (nreverse onelines))
+Multi-line canidates are merged into a single line."
+  (let* ((single-line-candidates ())
+
+         ;; The indicators are the same for all multi-line candidates, and so
+         ;; only need to be gotten from `selectrum-multiline-display-settings'
+         ;; once.
+         (selectrum--match-transformation
+          (alist-get 'match selectrum-multiline-display-settings))
+         (selectrum--match-display
+          (car selectrum--match-transformation))
+         (selectrum--match-face
+          (cadr selectrum--match-transformation))
+         (selectrum--truncation-transformation
+          (alist-get 'truncation selectrum-multiline-display-settings))
+         (selectrum--truncation-display
+          (car selectrum--truncation-transformation))
+         (selectrum--truncation-face
+          (cadr selectrum--truncation-transformation))
+         (selectrum--newline-transformation
+          (alist-get 'newline selectrum-multiline-display-settings))
+         (selectrum--newline-display
+          (car selectrum--newline-transformation))
+         (selectrum--newline-face
+          (cadr selectrum--newline-transformation))
+         (selectrum--whitespace-transformation
+          (alist-get 'whitespace selectrum-multiline-display-settings))
+         (selectrum--whitespace-display
+          (car selectrum--whitespace-transformation))
+         (selectrum--whitespace-face
+          (cadr selectrum--whitespace-transformation)))
```

Comment:
Thanks for renaming those variables. Having the selectrum prefix for non local variables is unusual, I haven't found a good rule how to avoid dashes and create longer names myself. Usually I try to abbreviate them like:

`selectrum--match-transformation` -> `mtransformation`

But if you want to keep the whole version you could also use `match/transformation`.